crates/common/src/deposit/mod.rs

```diff
@@ -1,15 +1,19 @@
 //! Tramsaction types for Optimism.
 
 mod source;
+use alloy_consensus::Transaction;
 pub use source::{
     DepositSourceDomain, DepositSourceDomainIdentifier, L1InfoDepositSource, UpgradeDepositSource,
     UserDepositSource,
 };
 
 use alloy_primitives::B256;
 
+/// Identifier for an Optimism deposit transaction
+pub const DEPOSIT_TX_TYPE_ID: u8 = 126; // 0x7E
+
 /// A trait representing a deposit transaction with specific attributes.
-pub trait DepositTransaction {
+pub trait DepositTransaction: Transaction {
```

Comment:
should this be renamed to OpTransaction?

Reply:
no this is just for deposit transaction. `OpTransaction` should be renamed `OpTransactionEnvelope` and have this as super trait if anything https://github.com/alloy-rs/alloy/pull/1910